### ISSUE TITLE
Error connecting to custom gitlab instance, login successful

### ISSUE THREAD

**Issue by Spreadcat** [profclems/glab#354]
I've got a problem using gitlab almost right from the start.
* I can login to the private gitlab instance using an access token.
* I can not list any issues of the repository I'm currently in-

**Description**

1. I login using `glab login auth` and the interactive method. That seems to work fine.
2. I try to list all the issue of the repo I'm currently in with my cli and I get the error message

   ```
   error connecting to gitlab
   check your internet connection or status.gitlab.com or 'Run sudo gitlab-ctl status' on your server if self-hosted
   ```

**Expected Behavior**

```
cd $gitrepo
gitrepo>$ glab auth login
? What GitLab instance do you want to log into? GitLab Self-hosted Instance
? GitLab hostname: gitlab.example.com
- Logging into gitlab.example.com

Tip: you can generate a Personal Access Token here https://gitlab.example.com/profile/personal_access_tokens
The minimum required scopes are 'api' and 'write_repository'.
? Paste your authentication token: ********************
? Choose default git protocol SSH
- glab config set -h gitlab.example.com git_protocol ssh
✓ Configured git protocol
✓ Logged in as $user

gitrepo>$ glab issue list -a
issue 1 .... 
issue 2 ...
```

**Actual Behavior**

```
cd $gitrepo
gitrepo>$ glab auth login
? What GitLab instance do you want to log into? GitLab Self-hosted Instance
? GitLab hostname: gitlab.example.com
- Logging into gitlab.example.com

Tip: you can generate a Personal Access Token here https://gitlab.example.com/profile/personal_access_tokens
The minimum required scopes are 'api' and 'write_repository'.
? Paste your authentication token: ********************
? Choose default git protocol SSH
- glab config set -h gitlab.example.com git_protocol ssh
✓ Configured git protocol
✓ Logged in as $user

gitrepo>$ glab issue list -a
error connecting to gitlab
check your internet connection or status.gitlab.com or 'Run sudo gitlab-ctl status' on your server if self-hosted
```

**Steps to Reproduce**

See example in actual behaviour.

**Context**

I was trying to test out glab for the first time in order to get an alternative to the GUI based workflow.

**Your Environment**
* Version used: 1.12.1
* Operating System and version: Fed 32
* Gitlab: GitLab Community Edition 13.6.1 

* I've got an ssh alias on the string "gitlab". `ssh gitlab` will connect to `gitlab.example.com` with an IdentifyFile that grants access. 

  ```
  # ~/.ssh/config
  ...
  host gitlab
    IdentityFile ~/.ssh/id_rsa
   user: git
  ```
  I didn't find any mentioning of glab--cli using the existing ssh config or not. Trying to configure just `gitlab` (like the host alias) fails.
* This works as well: `telnet gitlab.example.com 22`.
* Connecting to gitlab.com works fine.

**Comment by profclems** [comment]
Thanks for reporting this issue @Spreadcat 

We do support custom SSH Configs because git supports them. For example, this will be supported by glab per your ssh config:

```sh
git remote add origin gitlab:profclems/glab
```
The remote URL will be interpreted as `ssh://gitlab.com/profclems/glab` per your ssh config. 

Here is our current ssh config parser:
https://github.com/profclems/glab/blob/trunk/internal/git/ssh_config.go

The only thing `glab` has issues with is support for ssh include directives

Does your config work well with git?

**Comment by profclems** [comment]
Here is our current ssh config parser:
https://github.com/profclems/glab/blob/trunk/internal/git/ssh_config.go

**Comment by Spreadcat** [comment]
> Does your config work well with git?

Yes, it does. with git I do not have issues cloning, pull or pushing to the repository.

I did some more testing.
It seems to have something to do with the use of the host aliases in the file `~/.ssh/config`.


----

This does not work with glab, but works with git:

```
$ git remote -v
origin gitlab:user/repo (fetch)
origin gitlab:user/repo (push)
```

```
# ~/.ssh/config

host gitlab
  IdentifyFile ....
  hostname gitlab.example.com
  user git
```

```
$ $ glab issue list -a
error connecting to gitlab
check your internet connection or status.gitlab.com or 'Run sudo gitlab-ctl status' on your server if self-hosted
```

----

This DOES work with glab and in git:

```
$ git remote -v
origin gitlab.example.com:user/repo (fetch)
origin gitlab.example.com:user/repo (push)
```

```
# ~/.ssh/config
host gitlab.example.com
  IdentifyFile ...
  user git
```

```
$ $ glab issue list
Showing 1 open issue in user/repo that match your search (Page 1)

#1  Test issue    about 1 day ago
```

**Comment by profclems** [comment]
@Spreadcat this should be resolved by the linked PR; pending release. You can however, build from trunk and  confirm it's been resolved

**Comment by Spreadcat** [comment]
I'm not sure how this is related to the example code above, where only the host parameter has been added to overwrite the hostname from origin, but I will wait the new release and test that after that. I don't expect it to be resolved, though.
Or could you confirm the error, reproduce it and solve it with the PR? @profclems

**Comment by profclems** [comment]
This is fully fixed by #501. The error was caused by `glab` not using the hostname specified but the host alias name directly.
